internal/util/test/crds.go

```diff
@@ -17,11 +19,27 @@ import (
 // Testing Utility Functions - CRDs
 // -----------------------------------------------------------------------------
 
-const (
-	kongCRDsKustomize = "../../config/crd/"
+var (
+	kongCRDsKustomize = "config/crd/"
+	crdsOnce          sync.Once
 )
 
 func DeployCRDsForCluster(ctx context.Context, cluster clusters.Cluster) error {
+	var err error
+	crdsOnce.Do(func() {
+		var dir string
+		// We need the repo root directory to be able to run this  from anywhere in the repository.
+		dir, err = getRepoRoot(ctx)
+		if err != nil {
+			return
+		}
+
+		kongCRDsKustomize = filepath.Join(dir, kongCRDsKustomize)
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
Wouldn't it be more direct to have a helper init function and the variable definition done like so?
```go
kongCRDsKustomize = initKongCRDsKustomize()
```

Reply:
With init this would be done regardless, whenever the package gets imported. With `sync.Once` it's done only the first time the enclosing functions gets called.

Reply:
🤔 Aren't globals initialized only once per executable? If the import graph looks like so 
```
main->A->Z
   \->B->Z
```

Z's initialization is run only once as I understand. It's mentioned e.g. [here](https://golangbyexample.com/init-function-golang/):

> Note here that package initialization is only done once even if it is imported several times.

Reply:
I see. 👍 I still think it's not worth the complexity associated with `sync.Once`. Please note the drawbacks of it:
- If the first call to `DeployCRDsForCluster` fails, it will not return an error in subsequent calls and will happily use the uninitialized `kongCRDsKustomize`
- `kongCRDsKustomize` before a call to `DeployCRDsForCluster` stays in an uninitialized state with `config/crd/` value
- There's one more global to keep - `crdsOnce`

With variable init, we get one call per test executable (which would happen anyway because we call `DeployCRDsForCluster` in the test suite) and less complexity. In the init function, we can panic if we cannot get the repo root because it's an unrecoverable state.

Reply:
That makes sense. Let's use `init for this`. PTAL https://github.com/Kong/kubernetes-ingress-controller/pull/4823/commits/ebc7f6a110d3707bbc853fe7d5a61f924db53e14